contracts/TelfordSource.sol

```diff
@@ -0,0 +1,96 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract TelfordSource {
+    // this contract will be deployed to Arbitrum
+    address payable private bonderAddress;
+    address private l1Relayer;
+
+    address private owner;
+
+    uint256 constant BONDER_FEE = 0.2 ether;
+
+    struct BridgeInfo {
+        address userAddress;
+        address bonderAddress;
+        uint256 bridgeAmount;
+        uint256 bonderPayment;
+    }
+
+    mapping(uint => BridgeInfo) private bridgeRequests;
+
+    event BridgeRequested(
+        address userAddress,
+        uint256 amount,
+        uint256 transferId
+    );
+    event BonderReimbursed(uint256 bonderPayment);
+
+    modifier onlyOwner() {
+        require(
+            msg.sender == owner,
+            "Sorry pal, I can only be called by the contract owner!"
+        );
+        _;
+    }
+
+    modifier onlyL1Relayer() {
+        require(
+            msg.sender == l1Relayer,
+            "Sorry pal, I can only be called by the L1Relayer!"
+        );
+        _;
+    }
+
+    constructor(address _bonderAddress, address _l1RelayerAddress) {
+        owner = msg.sender;
+        bonderAddress = payable(_bonderAddress);
+        l1Relayer = _l1RelayerAddress;
+    }
+
+    function bridge(uint256 _transferId) external payable {
```

Comment:
the `_transferId` should be set by the contract (and not the front end app). the FE app doesn't maintain any state.